### ISSUE TITLE
Deserialization doesn't support Serde alternative enum representations

### ISSUE THREAD

**Issue by StyMaar** [gura-conf/serde-gura#4]
Serde allows different enum representations: `Internally tagged`, `Adjacently tagged` and `Untagged` in addition to the default `Externally tagged` representation [see Enum representations in Serde's doc](https://serde.rs/enum-representations.html).

Serde gura properly support **seralization** of such enum representation, but fails to perform **deserialization** of such representation.

Example:

### Internally tagged

```rust

use serde::{Deserialize, Serialize};
use serde_gura::Result;

#[derive(Serialize, Deserialize, PartialEq, Debug)]
#[serde(rename_all = "lowercase",tag = "type")]
enum MyEnum{
    Database(Database)
}

#[derive(Serialize, Deserialize, PartialEq, Debug)]
struct Database {
    ip: String,
    port: Vec<u16>,
    connection_max: u32,
    enabled: bool,
}

fn main() -> Result<()> {


    // You have some type.
    let database = MyEnum::Database(Database {
        ip: "127.0.0.1".to_string(),
        port: vec![80, 8080],
        connection_max: 1200,
        enabled: true,
    });

    let sss_str = serde_gura::to_string(&database)?;
    
    println!("{}", &sss_str);
 

    // Deserialize it back to a Rust type
    let deserialized_database: MyEnum = serde_gura::from_str(&sss_str)?;
    
    println!("{:#?}", deserialized_database);

    Ok(())
}
```

```
type: "database"
ip: "127.0.0.1"
port: [80, 8080]
connection_max: 1200
enabled: true
Error: ExpectedIdentifier
```

### Untagged

```rust
use serde::{Deserialize, Serialize};
use serde_gura::Result;

#[derive(Serialize, Deserialize, PartialEq, Debug)]
#[serde(rename_all = "lowercase",untagged)]
enum MyEnum{
    Database(Database)
}

#[derive(Serialize, Deserialize, PartialEq, Debug)]
struct Database {
    ip: String,
    port: Vec<u16>,
    connection_max: u32,
    enabled: bool,
}

fn main() -> Result<()> {


    // You have some type.
    let database = MyEnum::Database(Database {
        ip: "127.0.0.1".to_string(),
        port: vec![80, 8080],
        connection_max: 1200,
        enabled: true,
    });

    let sss_str = serde_gura::to_string(&database)?;
    
    println!("{}", &sss_str);
 

    // Deserialize it back to a Rust type
    let deserialized_database: MyEnum = serde_gura::from_str(&sss_str)?;
    
    println!("{:#?}", deserialized_database);

    Ok(())
}
```

output

```
ip: "127.0.0.1"
port: [80, 8080]
connection_max: 1200
enabled: true
Error: InvalidType
```

### Adjacently tagged

```rust

use serde::{Deserialize, Serialize};
use serde_gura::Result;

#[derive(Serialize, Deserialize, PartialEq, Debug)]
#[serde(rename_all = "lowercase",tag = "t", content = "c")]
enum MyEnum{
    Database(Database)
}

#[derive(Serialize, Deserialize, PartialEq, Debug)]
struct Database {
    ip: String,
    port: Vec<u16>,
    connection_max: u32,
    enabled: bool,
}

fn main() -> Result<()> {


    // You have some type.
    let database = MyEnum::Database(Database {
        ip: "127.0.0.1".to_string(),
        port: vec![80, 8080],
        connection_max: 1200,
        enabled: true,
    });

    let sss_str = serde_gura::to_string(&database)?;
    
    println!("{}", &sss_str);
 

    // Deserialize it back to a Rust type
    let deserialized_database: MyEnum = serde_gura::from_str(&sss_str)?;
    
    println!("{:#?}", deserialized_database);

    Ok(())
}
```

output
```
t: "database"
c:
    ip: "127.0.0.1"
    port: [80, 8080]
    connection_max: 1200
    enabled: true
Error: ExpectedIdentifier
```

**Comment by Genarito** [comment]
Thank you for the report! I'll analyze the problem and publish the fix along with the changes listed in #3 

Regards!

**Comment by Genarito** [comment]
All the 3 enums alternatives reported are now supported in [version 0.1.3](https://crates.io/crates/serde_gura). This issue also allowed me to find some little mistakes that I took the opportunity to correct :ok_hand: 

Thanks for making Gura better!
Best regards!
